libraries/node-core-library/src/Terminal/Terminal.ts

```diff
@@ -23,6 +23,18 @@ export class Terminal {
   public constructor(provider: ITerminalProvider) {
     this._providers = new Set<ITerminalProvider>();
     this._providers.add(provider);
+
+    // Bind the public members to "this"
+    this.registerProvider = this.registerProvider.bind(this);
+    this.unregisterProvider = this.unregisterProvider.bind(this);
+    this.write = this.write.bind(this);
+    this.writeLine = this.writeLine.bind(this);
+    this.writeWarning = this.writeWarning.bind(this);
+    this.writeWarningLine = this.writeWarningLine.bind(this);
+    this.writeError = this.writeError.bind(this);
+    this.writeErrorLine = this.writeErrorLine.bind(this);
+    this.writeVerbose = this.writeVerbose.bind(this);
```

Comment:
I kind of disagree with this:  If it a makes sense to autobind writeLine(), then we need a PR to autobind every single public API in our entire code base.  But that would need to be automated -- it's too difficult to manage manually.